docs/architecture/assessment-runtime/results-collector.md

```diff
@@ -1 +1,34 @@
-# Results Collector
\ No newline at end of file
+# Results Collector
+
+The Results Collector is responsible for collecting all the observations from the Assessment Action Jobs and processing them to generate metrics and other analysis results.
+
+1. The flow begins with the **Job Manager** sending a request to the ****Job Runner**** to run a specific job.
+2. Upon receiving the request, the ****Job Runner**** proceeds to provide the necessary context to the **Assessment Action Job**.
+3. The **Assessment Action Job** is executed and runs the compliance checks against the target component.
+4. The **Assessment Action Job** then passes these observations to the **Results Collector**.
+5. The **Results Collector** takes the observations and processes them internally. This may involve interpreting, analyzing, or organizing the information.
+6. The **Results Collector** continues its internal operations by generating metrics based on the processed observations. These metrics could represent various measurements or analytics related to the assessment.
```

Comment:
I agree with the flow, but wanted to ask if we have an idea of how the metrics are going to look like if it is generated by the Results Collector. Or are we thinking here of raw metrics (Observations, Findings, Risks and Attestations ?)

Also not related with the PR but to start the discussion: generally speaking I think we might want to have some spec in the configuration service for these (i.e. to tell to the Results Collector how to generate a risk whenever the Logs contain X, how to generate an observation from the AssessmentAction Job output, and what's the Attestation logic (this one we could probably just infer based on the other two).

Reply:
> I agree with the flow, but wanted to ask if we have an idea of how the metrics are going to look like if it is generated by the Results Collector. Or are we thinking here of raw metrics (Observations, Findings, Risks and Attestations ?)

I think we need to have separate sessions specifically for metrics. My initial idea was to collect as much information as possible from different layers of the runtime, such as control ID, component ID, assessment ID, assessment action, etc. In this way, we can manipulate the data as we see fit. We might also need to change the naming since the actual collection is handled by the JobRunner. "Results Collector" might be transformed into "Results Publisher" to push the events back to the event bus.

Also, we need to decide which concept of OSCAL is most suitable for metrics collection in terms of terminology. I believe that "Observation" is the closest concept, as it holds empirical and perhaps measurable data. "Findings" and "Risk" are more open to human interpretation.

> Also not related with the PR but to start the discussion: generally speaking I think we might want to have some spec in the configuration service for these (i.e. to tell to the Results Collector how to generate a risk whenever the Logs contain X, how to generate an observation from the AssessmentAction Job output, and what's the Attestation logic (this one we could probably just infer based on the other two).

You might want to create another issue for this one in the main project to keep it around :)

Reply:
I agree that Observation is what is going to be collected, and we should support business rules to automatically fill in findings, risks and attestation, whenever possible 😄 I start to see how everything is connected, and I'm happy about it.

I'm KISS'ing a bit and just adding an issue that maps the "Add processing metrics entity" in our issues, just so we remember it.